integration/integration_test.go

```diff
@@ -4039,8 +4039,21 @@ func testDiscovery(t *testing.T, suite *integrationTestSuite) {
 	helpers.WaitForActiveTunnelConnections(t, main.Tunnel, "cluster-remote", 1)
 	helpers.WaitForActiveTunnelConnections(t, secondProxy, "cluster-remote", 1)
 
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
```

Comment:
```suggestion
	require.EventuallyWithT(t, func(t *assert.CollectT) {
```

WDYT about shadowing `t` so that it's impossible to mistakenly assert against the wrong thing inside the closure?